benches/helpers.rs

```diff
@@ -146,9 +146,13 @@ pub async fn ws_server(handle: tokio::runtime::Handle) -> (String, jsonrpsee::se
 	let mut module = gen_rpc_module();
 
 	module
-		.register_subscription(SUB_METHOD_NAME, SUB_METHOD_NAME, UNSUB_METHOD_NAME, |_params, mut sink, _ctx| {
+		.register_subscription(SUB_METHOD_NAME, SUB_METHOD_NAME, UNSUB_METHOD_NAME, |_params, pending, _ctx| {
 			let x = "Hello";
-			tokio::spawn(async move { sink.send(&x) });
+			tokio::spawn(async move {
+				let sink = pending.accept().await.unwrap();
+				let msg = sink.build_message(&x).unwrap();
```

Comment:
This is a bit awkward but I think it's easier to reason about why send failed now.

We could keep to old behavior which serializes the message to JSON-RPC notification.

Reply:
The reason why `send` takes `JSON string` is that if the send fails we return the message and that's a JSON string.

Reply:
It does feel to me like this means that the "easy path" allows for a user to forget to use the `build_message` etc and send garbage back.

I feel like the two options might be:

1. If there are different message types we want the user to be able to build, then have `build_message` and `build_error_message` return some opaque type that `sink.send` requires (to enforce that the user builds a valid message), or
2. If it's just the two types, maybe having a `sink.send` and a `sink.send_error` that take a `T: Serialize` and ensure well formed messages.

Reply:
It does feel to me like this means that the "easy path" allows for a user to forget to use the `build_message` etc and send garbage back.

I feel like the two options might be:

1. If there are different message types we want the user to be able to build, then have `build_message` and `build_error_message` return some opaque type that `sink.send`/`sink.try_send` requires (to enforce that the user builds a valid message), or
2. If it's just the two types, maybe having a `sink.send` and a `sink.send_error` that take a `T: Serialize` and ensure well formed messages (though I can see try_send now, so I think option 1 makes more sense offhand).

Reply:
Yeah, I was thinking about having a wrapper that we call `build_message() -> SubscriptionMessage(String)` which forces everyone to use `build_message` explicitly because access to the underlying String is not possible.

Then errors would return that as well but it's a bit annoying to use anyway. 

However, if we provide `encode_and_send` or something as I added then to user can select what use.

Reply:
Mmm, how often would a user want to call `build_error_message`? I guess the advantage of keeping building and sending separate is if you want to build differently (ie Ok message or Error message) and then send differently (ie send or try_send). But I did just see those `encode_and_send` fns and they could be an addition to the API for the most common case(s) (ie building an Ok message and `send().await`ing it; I'm not really sure when one would want to use `try_send` offhand; I guess just for epheremal things that they don't care about succeeding or not..)

Reply:
I had a look to introduce `SubscriptionMessage::new(T) > Result<SubscriptionMessage(String), Error>` but it was quite messy to support as the the `send API` returns back the message on result 

So, I kept this for now i.e, build_message :(